[WTF] Prepare for InstallAPI by hiding project headers and fixing inconsistencies

emw-apple opened the pull request:
#### a10d22f7b1fb715a33f3c1293c49916497e34bb3
<pre>
[WTF] Prepare for InstallAPI by hiding project headers and fixing inconsistencies
<a href="https://bugs.webkit.org/show_bug.cgi?id=254614">https://bugs.webkit.org/show_bug.cgi?id=254614</a>
rdar://107332243

Reviewed by Alexey Proskuryakov.

As part of the larger project of JavaScriptCore InstallAPI on Apple
platforms, we need to prepare WTF&apos;s headers to be parsable by TAPI.
Make small non-functional changes to headers which fix symbol visibility
inconsistences, hide headers not used by clients, and generally make the
headers parsable by TAPI.

* Source/WTF/WTF.xcodeproj/project.pbxproj: Stop installing some headers
  to /usr/local/include/wtf which are only used for compiling WTF. This
  avoids some TAPI issues, especially with headers that are not valid
  Objective-C++ and therefore arguably never belonged in WTF&apos;s private
  header set.
* Source/WTF/wtf/AutodrainedPool.h: Allow TAPI to parse this even though
  it uses an Objective-C frontend.
* Source/WTF/wtf/SingleRootGraph.h: Add missing include.
* Source/WTF/wtf/Threading.cpp:
(WTF::initializeThreading): Deleted, was a compatibility hack to support
  Safari from many years ago.
* Source/WTF/wtf/WallTime.h: Remove unimplemented sleep() declaration.
* Source/WTF/wtf/WorkQueue.h: Remove unimplemented assertIsCurrent()
  declaration.
* Source/WTF/wtf/ios/WebCoreThread.h: Add the same #if guard on these
  declarations that is used in WebCoreThread.cpp.
* Source/WTF/wtf/text/WTFString.h: Remove unimplemented
  numberToStringFixedWidth() declaration.

Of the headers no longer being installed, fast_float headers need
updating to use quote-style impots. This is actually how they are
imported upstream.

* Source/WTF/wtf/FastFloat.cpp: Use quote-imports.
* Source/WTF/wtf/fast_float/ascii_number.h: Ditto.
* Source/WTF/wtf/fast_float/bigint.h: Ditto.
* Source/WTF/wtf/fast_float/decimal_to_binary.h: Ditto.
* Source/WTF/wtf/fast_float/digit_comparison.h: Ditto.
* Source/WTF/wtf/fast_float/fast_float.h: Ditto.
* Source/WTF/wtf/fast_float/parse_number.h: Ditto.
* Source/WTF/wtf/fast_float/simple_decimal_conversion.h: Ditto.

Fix symbol inconsistencies related to use of OS objects, which are
declared as opaque NSObjects when using an Objective-C compiler, and are
otherwise C struct typedefs. When WTF&apos;s implementation of an OS object
function is in C++, the mangled symbol name does not match the
declaration an Objective-C++ compiler sees. This means that these
interfaces were never callable from Objective-C++. There are two such
inconsistencies detected by TAPI.

1. OSLogPrintStream, which uses os_log_type_t. Since this class is only
   applicable to Darwin-based platforms, change its implementation to
   Objective-C++ and move it to wtf/darwin/.

* Source/JavaScriptCore/runtime/Options.cpp: Update include directive.
* Source/JavaScriptCore/tools/Integrity.cpp: Update include directive.
* Source/WTF/wtf/CMakeLists.txt: Remove OSLogPrintStream from headers
  and sources, since they do nothing on non-Apple platforms.
* Source/WTF/wtf/darwin/OSLogPrintStream.h: Renamed from Source/WTF/wtf/OSLogPrintStream.h.
* Source/WTF/wtf/darwin/OSLogPrintStream.mm: Renamed from Source/WTF/wtf/OSLogPrintStream.cpp.

2. MemoryPressureHandler::setDispatchQueue, which uses dispatch_queue_t.
   While it is obviously Darwin-only, setDispatchQueue is called by C++
   cross-platform C++ code (jsc.cpp), so moving the implementation to a
   .mm file breaks this use site. Instead, since it&apos;s just a single
   assignment, make it an inline function.

* Source/WTF/wtf/MemoryPressureHandler.cpp:
(WTF::MemoryPressureHandler::setDispatchQueue): Deleted.
* Source/WTF/wtf/MemoryPressureHandler.h:
(WTF::MemoryPressureHandler::setDispatchQueue):

Canonical link: <a href="https://commits.webkit.org/262668@main">https://commits.webkit.org/262668@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/360bedb24c9d0d85a1db12839ded5ad636f25eeb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2203 "14 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2236 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2304 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/3127 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2245 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2176 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2317 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2282 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1996 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2223 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1975 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1980 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2983 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/1954 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1957 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1854 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/1844 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/2034 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/2004 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/3157 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/2110 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2022 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1816 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/2265 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1948 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1966 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/518 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/547 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2139 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/2315 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/633 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WTF/wtf/darwin/OSLogPrintStream.mm:
```diff
@@ -24,7 +24,7 @@
 
 #include "config.h"
-#include <wtf/OSLogPrintStream.h>
+#include <wtf/darwin/OSLogPrintStream.h>
```
Surprised that this is still a framework style include.

emw-apple replied:
This does feel like it would be more appropriate as a quoted include, since it's the companion header for this implementation file. I'll change it.

emw-apple committed:
[WTF] Prepare for InstallAPI by hiding project headers and fixing inconsistencies
https://bugs.webkit.org/show_bug.cgi?id=254614
rdar://107332243

Reviewed by Alexey Proskuryakov.

As part of the larger project of JavaScriptCore InstallAPI on Apple
platforms, we need to prepare WTF's headers to be parsable by TAPI.
Make small non-functional changes to headers which fix symbol visibility
inconsistences, hide headers not used by clients, and generally make the
headers parsable by TAPI.

* Source/WTF/WTF.xcodeproj/project.pbxproj: Stop installing some headers
  to /usr/local/include/wtf which are only used for compiling WTF. This
  avoids some TAPI issues, especially with headers that are not valid
  Objective-C++ and therefore arguably never belonged in WTF's private
  header set.
* Source/WTF/wtf/AutodrainedPool.h: Allow TAPI to parse this even though
  it uses an Objective-C frontend.
* Source/WTF/wtf/SingleRootGraph.h: Add missing include.
* Source/WTF/wtf/Threading.cpp:
(WTF::initializeThreading): Deleted, was a compatibility hack to support
  Safari from many years ago.
* Source/WTF/wtf/WallTime.h: Remove unimplemented sleep() declaration.
* Source/WTF/wtf/WorkQueue.h: Remove unimplemented assertIsCurrent()
  declaration.
* Source/WTF/wtf/ios/WebCoreThread.h: Add the same #if guard on these
  declarations that is used in WebCoreThread.cpp.
* Source/WTF/wtf/text/WTFString.h: Remove unimplemented
  numberToStringFixedWidth() declaration.

Of the headers no longer being installed, fast_float headers need
updating to use quote-style impots. This is actually how they are
imported upstream.

* Source/WTF/wtf/FastFloat.cpp: Use quote-imports.
* Source/WTF/wtf/fast_float/ascii_number.h: Ditto.
* Source/WTF/wtf/fast_float/bigint.h: Ditto.
* Source/WTF/wtf/fast_float/decimal_to_binary.h: Ditto.
* Source/WTF/wtf/fast_float/digit_comparison.h: Ditto.
* Source/WTF/wtf/fast_float/fast_float.h: Ditto.
* Source/WTF/wtf/fast_float/parse_number.h: Ditto.
* Source/WTF/wtf/fast_float/simple_decimal_conversion.h: Ditto.

Fix symbol inconsistencies related to use of OS objects, which are
declared as opaque NSObjects when using an Objective-C compiler, and are
otherwise C struct typedefs. When WTF's implementation of an OS object
function is in C++, the mangled symbol name does not match the
declaration an Objective-C++ compiler sees. This means that these
interfaces were never callable from Objective-C++. There are two such
inconsistencies detected by TAPI.

1. OSLogPrintStream, which uses os_log_type_t. Since this class is only
   applicable to Darwin-based platforms, change its implementation to
   Objective-C++ and move it to wtf/darwin/.

* Source/JavaScriptCore/runtime/Options.cpp: Update include directive.
* Source/JavaScriptCore/tools/Integrity.cpp: Update include directive.
* Source/WTF/wtf/CMakeLists.txt: Remove OSLogPrintStream from headers
  and sources, since they do nothing on non-Apple platforms.
* Source/WTF/wtf/darwin/OSLogPrintStream.h: Renamed from Source/WTF/wtf/OSLogPrintStream.h.
* Source/WTF/wtf/darwin/OSLogPrintStream.mm: Renamed from Source/WTF/wtf/OSLogPrintStream.cpp.

2. MemoryPressureHandler::setDispatchQueue, which uses dispatch_queue_t.
   While it is obviously Darwin-only, setDispatchQueue is called by C++
   cross-platform C++ code (jsc.cpp), so moving the implementation to a
   .mm file breaks this use site. Instead, since it's just a single
   assignment, make it an inline function.

* Source/WTF/wtf/MemoryPressureHandler.cpp:
(WTF::MemoryPressureHandler::setDispatchQueue): Deleted.
* Source/WTF/wtf/MemoryPressureHandler.h:
(WTF::MemoryPressureHandler::setDispatchQueue):

Canonical link: https://commits.webkit.org/262668@main
